src/rust/storage/seg/src/hashtable/mod.rs

```diff
@@ -152,15 +159,224 @@ impl HashTable {
+            file_data[offset..end].copy_from_slice(bytes);
+
+            // --------------------- Store `mask` -----------------
+            offset += u64_size;
+            end += u64_size;
+
+            // cast `mask` to byte pointer
+            let byte_ptr = (&self.mask as *const u64) as *const u8;
+
+            // get corresponding bytes from byte pointer
+            let bytes = unsafe { ::std::slice::from_raw_parts(byte_ptr, u64_size) };
+
+            // store `mask` back to mmapped file
+            file_data[offset..end].copy_from_slice(bytes);
+
+            // --------------------- Store `data` -----------------
+            offset += u64_size;
+            end += buckets_size;
+
+            // for every `HashBucket`
+            for id in 0..total_buckets {
+                let begin = offset + (bucket_size as usize * id);
+                let finish = begin + bucket_size as usize;
+
+                // cast `HashBucket` to byte pointer
+                let byte_ptr = (&self.data[id] as *const HashBucket) as *const u8;
+
+                // get corresponding bytes from byte pointer
+                let bytes = unsafe { ::std::slice::from_raw_parts(byte_ptr, bucket_size) };
+
+                // store `HashBucket` back to mmapped file
+                file_data[begin..finish].copy_from_slice(bytes);
+            }
+
+            // --------------------- Store `started` -----------------
+            offset += buckets_size;
+            end += started_size;
+
+            // cast `started` to byte pointer
+            let byte_ptr = (&self.started as *const Instant) as *const u8;
+
+            // get corresponding bytes from byte pointer
+            let bytes = unsafe { ::std::slice::from_raw_parts(byte_ptr, started_size) };
+
+            // store `started` back to mmapped file
+            file_data[offset..end].copy_from_slice(bytes);
+
+            // --------------------- Store `next_to_chain` -----------------
+            offset += started_size;
+            end += u64_size;
+
+            // cast `next_to_chain` to byte pointer
+            let byte_ptr = (&self.next_to_chain as *const u64) as *const u8;
+
+            // get corresponding bytes from byte pointer
+            let bytes = unsafe { ::std::slice::from_raw_parts(byte_ptr, u64_size) };
+
+            // store `next_to_chain` back to mmapped file
+            file_data[offset..end].copy_from_slice(bytes);
+
+            gracefully_shutdown = true;
+
+            // TODO: check if this flushes the CPU caches
+            pool.flush()
+                .expect("failed to flush `HashTable` to storage");
+        }
+        gracefully_shutdown
+    }
+
     /// Lookup an item by key and return it
     pub fn get(&mut self, key: &[u8], segments: &mut Segments) -> Option<Item> {
         let hash = self.hash(key);
         let tag = tag_from_hash(hash);
         let bucket_id = hash & self.mask;
 
+        // ccc: get bucket corresponding to the key
```

Comment:
I assume these comments were for your own notes as you familiarized yourself with the code. Please remove them